roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -165,17 +176,35 @@ def is_available_skopeo_image(self, image, default_registries):
         """Use Skopeo to determine if required image exists in known registry(s)."""
         registries = default_registries
 
-        # if image already includes a registry, only use that
+        # If image already includes a registry, only use that.
+        # NOTE: This logic would incorrectly identify images that do not use a namespace, e.g.
+        # registry.access.redhat.com/rhel7 as if the registry were a namespace.
+        # It's not clear that there's any way to distinguish them, but fortunately
+        # the current set of images all look like [registry/]namespace/name[:version].
         if image.count("/") > 1:
             registry, image = image.split("/", 1)
             registries = [registry]
 
         for registry in registries:
-            args = {
-                "_raw_params": self.skopeo_img_check_command + " docker://{}/{}".format(registry, image)
-            }
-            result = self.execute_module("command", args)
+            if registry not in self.can_reach_registry:
+                self.can_reach_registry[registry] = self.connect_to_registry(registry)
+            if not self.can_reach_registry[registry]:
+                continue
+
+            args = {"_raw_params": self.skopeo_img_check_command.format(registry=registry, image=image)}
+            result = self.execute_module_with_retries("command", args)
             if result.get("rc", 0) == 0 and not result.get("failed"):
                 return True
+            if result.get("rc") == 124:  # RC 124 == timed out; mark unreachable
+                self.can_reach_registry[registry] = False
 
         return False
+
+    def connect_to_registry(self, registry):
+        """Use ansible wait_for module to test connectivity from host to registry. Returns bool."""
+        # test a simple TCP connection
+        (host, _, port) = registry.partition(":")
```

Comment:
unnecessary left parens?

Reply:
Note that if ":" not in registry, then host and port will be empty strings.

Reply:
Yes I guess the parens are superfluous.

Lack of `:` is handled... actually the whole string is put in `host` in that case and port defaults to 443. You think it needs a clarifying comment?

Reply:
> Lack of : is handled... actually the whole string is put in host in that case and port defaults to 443. You think it needs a clarifying comment?

Right, I misinterpreted `str.partition`'s docstring. No need for comment.